tests/unit/tf/horovod/test_embedding.py

```diff
@@ -20,87 +13,22 @@ class TestSOKEmbedding:
-                local_size += 1
-            indices = np.arange(local_size) * hvd.size() + hvd.rank()
-            indices = tf.convert_to_tensor(indices, dtype=tf.int64)
-            local_indices.append(indices)
-        out1 = []
-        for i in range(len(sok_vars)):
-            out1.append(tf.nn.embedding_lookup(sok_vars[i], local_indices[i]))
-
-        tf_vars = [
-            tf.Variable(tf.constant(initial_vals[i], shape=[rows[i], cols[i]], dtype=tf.float32))
-            for i in range(len(rows))
-        ]
-        out2 = []
-        for i, v in enumerate(tf_vars):
-            out2.append(tf.nn.embedding_lookup(v, local_indices[i]))
-
-        # Check results
-        diff = 0
-        for i in range(len(out1)):
-            length = out1[i] ** 2 + out2[i] ** 2 + 1e-8
-            diff = diff + tf.reduce_sum((out1[i] - out2[i]) ** 2 / length)
-        print("[SOK INFO] diff:", diff)
-        assert diff < 1e-6
-
-    @pytest.mark.parametrize("dim", [16, 32])
-    def test_distributed_variables(self, dim):
-        rows = [65536 * 10, 65536]
-        cols = [128, 4]
-
-        # initial value of embedding table
-        weights = []
-        for i in range(len(rows)):
-            weight = np.random.rand(rows[i], cols[i]).astype(np.float32)
-            weight = tf.convert_to_tensor(weight, dtype=tf.float32)
-            # make sure the weight is same on each rank
-            weight = hvd.allreduce(weight)
-            weights.append(weight)
-
-        # sok variables
-        sok_vars = [sok.Variable(w) for w in weights]
-        local_indices = []
-        for row in rows:
-            local_size = row // hvd.size()
-            if hvd.rank() < row % hvd.size():
-                local_size += 1
-            indices = np.arange(local_size) * hvd.size() + hvd.rank()
-            indices = tf.convert_to_tensor(indices, dtype=tf.int64)
-            local_indices.append(indices)
-
-        out1 = sok_vars
-        tf_vars = [tf.Variable(w) for w in weights]
-        out2 = []
-        for i, v in enumerate(tf_vars):
-            out2.append(tf.nn.embedding_lookup(v, local_indices[i]))
-
-        # Check results
-        diff = 0
-        for i in range(len(out1)):
-            length = out1[i] ** 2 + out2[i] ** 2 + 1e-8
-            diff = diff + tf.reduce_sum((out1[i] - out2[i]) ** 2 / length)
-        print("[SOK INFO] diff:", diff)
-        assert diff < 1e-6
+    def test_sok_embedding_basic(self):
+        embedding = SOKEmbedding(16, self.sample_column_schema, vocab_sizes=[10])
+        inputs = [tf.ragged.constant([[0, 1, 0], [1, 0]])]
+        combiners = ["sum"]
+        outputs = embedding(inputs, combiners)
+        assert outputs[0].shape == (2, 16)
+
+    def test_sok_embedding_pretrained(self):
+        weights = np.random.rand(10, 16)
+        embedding = SOKEmbedding.from_pretrained(16, [weights])
+        inputs = [tf.ragged.constant([[0, 1, 0], [1, 0]])]
+        combiners = ["sum"]
```

Comment:
I think it would also make sense to have a sensible default for `combiners` and make this an optional parameter.

Reply:
@edknv  Make sense. I will modify it

Reply:
Thanks. We can merge this PR to the main PR and apply the changes there.